specs/core/0_beacon-chain.md

```diff
@@ -1722,15 +1726,23 @@ def update_validator_registry(state: BeaconState) -> None:
     # Exit validators within the allowable balance churn
     balance_churn = 0
     for index, validator in enumerate(state.validator_registry):
-        if validator.status == ACTIVE_PENDING_EXIT:
+        if validator.status == ACTIVE_PENDING_NO_PENALTY_EXIT:
             # Check the balance churn would be within the allowance
             balance_churn += get_effective_balance(state, index)
             if balance_churn > max_balance_churn:
                 break
 
             # Exit validator
             update_validator_status(state, index, new_status=EXITED_WITHOUT_PENALTY)
-
+            
+    # Exit penalized validators, more than one epoch after they were penalized
+    for index, validator in enumerate(state.validator_registry):
+        is_delaying_exit = (
+            validator.status == ACTIVE_PENDING_PENALTY_EXIT and
+            state.slot > validator.latest_status_change_slot + MAX_SEED_LOOKAHEAD
```

Comment:
Check: `>` or `>=`?
We currently use `>=` for the boundary `ZERO_BALANCE_VALIDATOR_TTL` and `MIN_VALIDATOR_WITHDRAWAL_TIME`.

```suggestion
            state.slot >= validator.latest_status_change_slot + MAX_SEED_LOOKAHEAD
```
As a side note, probably good to align these constant names.